src/store/wizardSlice.ts

```diff
@@ -43,6 +44,18 @@ export type RegistrationType =
 
 export type ComplianceType = 'openscap' | 'compliance';
 
+export type UserWithAdditionalInfo = Required<User>;
+
+type UserPayload = {
+  index: number;
+  name: string;
+};
+
+type UserPasswordPayload = {
+  index: number;
+  password: string | undefined;
```

Comment:
Here I think that we need to think about what the empty state for the <TextInput> we will use for the password is. Is it `''` or `undefined`?

That makes working with the <TextInput> component in the React component code easier since we don't have to handle the special empty case by switching the value to undefined. On the other hand, it does mean  that when we generate the request we send to the image-builder API that we need to take care to convert the `''` to `undefined`. 

I don't think that one way is necessarily right or wrong, so here we should just let consistency guide us.

To be consistent with the rest of the code base, we should use `''` as the empty state (assuming things haven't changed too drastically since we made this decision over a year ago...).

So I think that the type for password should just be `string`.